modules/ingestor/src/service/format.rs

```diff
@@ -230,14 +230,10 @@ where
         let stream = pin!(stream);
         let stream = stream.map_err(|e| io::Error::new(io::ErrorKind::Other, format!("{e:?}")));
         let reader = SyncIoBridge::new(StreamReader::new(stream));
-        match flavor {
-            StreamFlavor::Json => info_span!("parse document")
-                .in_scope(|| serde_json::from_reader(reader))
-                .map_err(Error::Json),
-            StreamFlavor::Yaml => info_span!("parse document")
-                .in_scope(|| serde_yml::from_reader(reader))
-                .map_err(Error::Yaml),
-        }
+        info_span!("parse document").in_scope(|| match flavor {
+            StreamFlavor::Yaml => serde_yml::from_reader(reader).map_err(Error::Yaml),
+            StreamFlavor::Json => serde_json::from_reader(reader).map_err(Error::Json),
+        })
```

Comment:
I feel like the construction of the bridge should be a part of the parsing span, so will this work?

```rust
    tokio::task::spawn_blocking(move || {
        info_span!("parse document").in_scope(|| {
            let stream = pin!(stream);
            let stream = stream.map_err(|e| io::Error::new(io::ErrorKind::Other, format!("{e:?}")));
            let reader = SyncIoBridge::new(StreamReader::new(stream));
            match flavor {
                StreamFlavor::Json => serde_json::from_reader(reader).map_err(Error::Json),
                StreamFlavor::Yaml => serde_yml::from_reader(reader).map_err(Error::Yaml),
            }
        })
    })
    .await?
```

Reply:
Nevermind. I changed my mind.

Reply:
Lemme try!

Reply:
Seems to.